src/components/explorer/ExplorerHeader.tsx

```diff
@@ -0,0 +1,332 @@
+          <thead className="text-16 text-white/60">
+            <tr>
+              <th className="font-weight-400 text-start">ID</th>
+              <th className="font-weight-400 text-end">Trades</th>
+            </tr>
+          </thead>
+          <tbody className="font-weight-500">
+            {strategiesLoading &&
+              [1, 2, 3].map((id) => (
+                <tr key={id}>
+                  <td>
+                    <Loading />
+                  </td>
+                  <td>
+                    <Loading />
+                  </td>
+                </tr>
+              ))}
+            {strategies.map(({ id, idDisplay, base, quote, trades }) => (
+              <tr key={id}>
+                <td>
+                  <Link
+                    to="/strategy/$id"
+                    params={{ id }}
+                    className="block w-full"
+                  >
+                    <div className="bg-background-700 flex gap-8 rounded px-8">
+                      <TokensOverlap tokens={[base, quote]} size={18} />
+                      {idDisplay}
+                    </div>
+                  </Link>
+                </td>
+                <td className="w-full py-8 text-end">
+                  <Link
+                    to="/strategy/$id"
+                    params={{ id }}
+                    className="block w-full"
+                  >
+                    {formatter.format(trades)}
+                  </Link>
+                </td>
+              </tr>
+            ))}
+          </tbody>
+        </table>
+      </article>
+    </header>
+  );
+};
+
+const Loading = () => (
+  <div className="h-[30px] animate-pulse p-4">
+    <div className="bg-background-700 h-full rounded"></div>
+  </div>
+);
+
+const formatter = new Intl.NumberFormat(undefined, {
+  maximumFractionDigits: 0,
+});
+
+interface TradesProps {
+  trades: number;
+}
+
+const Trades = ({ trades }: TradesProps) => {
+  const ref = useRef<HTMLParagraphElement>(null);
+  const anims = useRef<Promise<Animation>[]>();
+  const lastTrades = useRef(0);
+  const initDelta = 60;
+
+  useEffect(() => {
+    let tradesChanged = false;
+    const start = async () => {
+      const from = lastTrades.current || trades - initDelta;
+      const to = trades;
+      const letters = ref.current!.children;
+      // Initial animation
+      if (!lastTrades.current) {
+        const initAnims: Promise<Animation>[] = [];
+        const next = formatter.format(from).split('');
+        for (let i = 0; i < next.length; i++) {
+          const v = next[i];
+          if (!'0123456789'.includes(v)) continue;
+          const anim = letters[i]?.animate(
+            [{ transform: `translateY(-${v}0%)` }],
+            {
+              duration: 1000,
+              delay: i * 100,
+              fill: 'forwards',
+              easing: 'cubic-bezier(1,-0.54,.65,1.46)',
+            }
+          );
+          initAnims.push(anim.finished);
```

Comment:
I'm getting an "anim is undefined" error from here sometimes as `letters[i]` is undefined and thus `anim` is undefined and `anim.finished` doesn't exist. Can you check it's not undefined?